Source/Utils/MsgLogger/MsgLoggerSerialization.H

```diff
@@ -0,0 +1,189 @@
+    * byte representation and it appends this vector at the end of an
+    * existing vector of chars (specialization of put_in_vec<T>).
+    *
+    * @tparam T the variable type
+    * @param[in] val a variable of type T to be serialized
+    * @param[in, out] vec a reference to the vector to which the byte representation of val is appended
+    */
+    template <>
+    inline void put_in_vec <char> (const std::vector<char>& val, std::vector<char>& vec)
+    {
+        put_in(static_cast<int>(val.size()), vec);
+        vec.insert(vec.end(), val.begin(), val.end());
+    }
+
+    /**
+    * This function extract a variable of type T from a byte vector, at the position
+    * given by a std::vector<char> iterator. The iterator is then advanced according to
+    * the number of bytes read from the byte vector. T must be either a trivially copyable type
+    * or an std::string (see specialization below).
+    *
+    * @tparam T the variable type (must be trivially copyable)
+    * @param[in, out] it the iterator to a byte vector
+    * @return the variable extracted from the byte array
+    */
+    template<typename T>
+    T get_out(std::vector<char>::const_iterator& it)
+    {
+        static_assert(std::is_trivially_copyable<T>(),
+            "Cannot extract non-trivally copyable types from char vectors,"
+            " with the exception of std::string.");
+
+        auto temp = std::array<char, sizeof(T)>{};
+        std::copy(it, it + sizeof(T), temp.begin());
+        it += sizeof(T);
+        T res;
+        std::memcpy(&res, temp.data(), sizeof(T));
+
+        return res;
+    }
+
+    /**
+    * This function extract an std::string from a byte vector, at the position
+    * given by a std::vector<char> iterator. The iterator is then advanced according to
+    * the number of bytes read from the byte vector. This is a specialization of
+    * get_out<T>
+    *
+    * @param[in, out] it the iterator to a byte vector
+    * @return the std::string extracted from the byte array
+    */
+    template<>
+    inline std::string get_out<std::string> (std::vector<char>::const_iterator& it)
+    {
+        const auto length = get_out<int> (it);
+        const auto str = std::string{it, it+length};
+        it += length;
+
+        return str;
+    }
+
+    /**
+    * This function extract an std::vector<T> from a byte vector, at the position
+    * given by a std::vector<char> iterator. The iterator is then advanced according to
+    * the number of bytes read from the byte vector. T must be either a trivially copyable type
+    * or an std::string.
+    *
+    * @tparam T the variable type (must be trivially copyable)
+    * @param[in, out] it the iterator to a byte vector
+    * @return the variable extracted from the byte array
+    */
+    template<typename T>
+    inline std::vector<T> get_out_vec (std::vector<char>::const_iterator& it)
+    {
+        static_assert(std::is_trivially_copyable<T>(),
```

Comment:
```suggestion
        static_assert(std::is_trivially_copyable<T>() || std::is_same<T,std::string>(),
```

Do we need that here as well?

Reply:
sure